projects/core/src/product/services/product-loading.service.ts

```diff
@@ -140,11 +148,19 @@ export class ProductLoadingService {
     // max age trigger add
     const maxAge = this.loadingScopes.getMaxAge('product', scope);
     if (maxAge && isPlatformBrowser(this.platformId)) {
-      const loadSuccess$ = this.actions$.pipe(
-        ofType(ProductActions.LOAD_PRODUCT_SUCCESS),
+      // we want to grab load product success and load product fail for this product and scope
+      const loadFinish$ = this.actions$.pipe(
         filter(
-          (action: ProductActions.LoadProductSuccess) =>
-            action.payload.code === productCode && action.meta.scope === scope
+          (
+            action:
+              | ProductActions.LoadProductSuccess
+              | ProductActions.LoadProductFail
+          ) =>
+            ((action.type === ProductActions.LOAD_PRODUCT_SUCCESS &&
+              action.payload.code === productCode) ||
+              (action.type === ProductActions.LOAD_PRODUCT_FAIL &&
+                action.payload === productCode)) &&
```

Comment:
I looked at the action and productCode is not in payload

Reply:
Good catch. thanks!